In [1]:
#Netflix 

In [74]:
#import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [93]:
#to read the .csv file
netflix_data = pd.read_csv('netflix1.csv')

In [95]:
#to read the first 3 data
netflix_data.head(3)

show_id     type                 title         director        country  \
0      s1    Movie  Dick Johnson Is Dead  Kirsten Johnson  United States   
1      s3  TV Show             Ganglands  Julien Leclercq         France   
2      s6  TV Show         Midnight Mass    Mike Flanagan  United States   

  date_added  release_year rating  duration  \
0  9/25/2021          2020  PG-13    90 min   
1  9/24/2021          2021  TV-MA  1 Season   
2  9/24/2021          2021  TV-MA  1 Season   

                                           listed_in  
0                                      Documentaries  
1  Crime TV Shows, International TV Shows, TV Act...  
2                 TV Dramas, TV Horror, TV Mysteries

In [97]:
#info() to get a  overview of the DataFrame, showing how many values are missing, and what type of data is present.
netflix_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8790 entries, 0 to 8789
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8790 non-null   object
 1   type          8790 non-null   object
 2   title         8790 non-null   object
 3   director      8790 non-null   object
 4   country       8790 non-null   object
 5   date_added    8790 non-null   object
 6   release_year  8790 non-null   int64 
 7   rating        8790 non-null   object
 8   duration      8790 non-null   object
 9   listed_in     8790 non-null   object
dtypes: int64(1), object(9)
memory usage: 686.8+ KB


As we can see there are no missing values in the following dataframes.

In [132]:
#to check the rows and coluns in 2d array
netflix_data.shape

(8790, 10)

In [136]:
#to check the data columns
netflix_data.columns.values

array(['show_id', 'type', 'title', 'director', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in'], dtype=object)

In [138]:
netflix_data.describe()

date_added  release_year
count                           8790   8790.000000
mean   2019-05-17 21:44:01.638225408   2014.183163
min              2008-01-01 00:00:00   1925.000000
25%              2018-04-06 00:00:00   2013.000000
50%              2019-07-03 00:00:00   2017.000000
75%              2020-08-19 18:00:00   2019.000000
max              2021-09-25 00:00:00   2021.000000
std                              NaN      8.825466

***Data Cleaning


Identify and handle missing data, correct data types and drop duplicates

In [101]:
#check for missing values
netflix_data.isnull().sum()

show_id         0
type            0
title           0
director        0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
dtype: int64

In [103]:
#to remove the duplicate values
netflix_data.drop_duplicates(inplace= True)

In [111]:
#Drop rows with missing cirtical info
netflix_data.dropna(subset=['director','country'], inplace=True)

In [116]:
!pip install wordcloud

In [120]:
from wordcloud import WordCloud

In [124]:
#convert 'date_added' to datetime
netflix_data['date_added'] = pd.to_datetime(netflix_data['date_added'])

In [130]:
#data types to confirm the changes
print(netflix_data.dtypes)

show_id                 object
type                    object
title                   object
director                object
country                 object
date_added      datetime64[ns]
release_year             int64
rating                  object
duration                object
listed_in               object
dtype: object


###Exploratory Data Analysis(EDA)

In [160]:
#Content Type Distribution(Movies vs TV shows)
#count the number of movies and tv shows
type_count=netflix_data['type'].value_counts()
type_count

type
Movie      6126
TV Show    2664
Name: count, dtype: int64

As we can see there are 6126 for movies and 2664 for TV shows